WhisperAPI/WhisperAPI/Services/Suggestions/SuggestionsService.cs

```diff
@@ -201,6 +219,39 @@ internal async Task<IEnumerable<Recommendation<Document>>> GetLastClickAnalytics
             }).OrderByDescending(recommendation => recommendation.Confidence);
         }
 
+        internal async Task<IEnumerable<Recommendation<Document>>> GetNearestDocumentsRecommendations(ConversationContext conversationContext, IEnumerable<string> documentsUri)
+        {
+            var contextEntities = new HashSet<string>(string.Join(" ", conversationContext.ContextItems.Where(x => x.Relevant).Select(c => c.NlpAnalysis.ParsedQuery)).Split(" "));
```

Comment:
line 224 too terse, please introduce variables or at least format the code with line breaks to be easier on the eyes